--- /home1/06959/mksn/.local/lib/python3.7/site-packages/torchvision/models/resnet.py
+++ /home1/06959/mksn/.local/lib/python3.7/site-packages/torchvision/models/resnet.py
@@ -1,5 +1,6 @@
 class BasicBlock(nn.Module):
     expansion = 1
+    __constants__ = ['downsample']
 
     def __init__(self, inplanes, planes, stride=1, downsample=None, groups=1,
                  base_width=64, dilation=1, norm_layer=None):